commands/src/main/java/org/wildfly/extras/creaper/commands/deployments/Deploy.java

```diff
@@ -0,0 +1,176 @@
+        step1.get(Constants.CONTENT).add(content);
+        if (serverGroupAddresses != null) {
+            for (ModelNode serverGroup : serverGroupAddresses) {
+                ModelNode sg = steps.add();
+                sg.set(getEmptyOperation(Constants.ADD, serverGroup));
+                sg.get(ENABLED).set(true);
+            }
+        }
+        return composite;
+    }
+
+    private ModelNode getEmptyOperation(String operationName, ModelNode address) {
+        ModelNode op = new ModelNode();
+        op.get(Constants.OP).set(operationName);
+        if (address != null) {
+            op.get(Constants.OP_ADDR).set(address);
+        } else {
+            // Just establish the standard structure; caller can fill in address later
+            op.get(Constants.OP_ADDR);
+        }
+        return op;
+    }
+
+
+    /**
+     * Builder for creating command for deploying deployment to the server.
+     */
+    public static final class Builder {
+        private File deploymentFile;
+        private InputStream deploymentInputStream;
+        private final boolean autoCloseInputStream;
+        private final String deploymentName;
+        private List<String> serverGroups;
+
+        public Builder(File deploymentFile) {
+            if (!deploymentFile.exists()) {
+                throw new IllegalArgumentException(deploymentFile + " doesn't exist!");
+            }
+            this.deploymentFile = deploymentFile;
+            this.deploymentName = deploymentFile.getName();
+            this.autoCloseInputStream = true;
+        }
+
+        public Builder(InputStream deploymentInputStream, String deploymentName,
+                       boolean autoCloseInputStream) {
+            this.deploymentInputStream = deploymentInputStream;
+            this.deploymentName = deploymentName;
+            this.autoCloseInputStream = autoCloseInputStream;
+        }
+
+        public Deploy build() {
+            InputStream inputStream = deploymentInputStream;
+            if (deploymentInputStream == null && deploymentFile != null) {
+                try {
+                    inputStream = new FileInputStream(deploymentFile);
+                } catch (FileNotFoundException ex) {
+                    throw new IllegalArgumentException(ex);
+                }
+            }
+            return new Deploy(inputStream, this);
+        }
+
+
+        /**
+         * <p>Domain mode only</p>
+         * <p>
+         *     Adds list of server groups for which the deployment should be applied.
+         * </p>
+         */
+        public Deploy toServerGroups(String... serverGroups) {
+            if (this.serverGroups == null && serverGroups != null) {
+                this.serverGroups = new ArrayList<String>();
+            }
+            if (serverGroups != null) {
+                this.serverGroups.addAll(Arrays.asList(serverGroups));
+            }
+            return build();
```

Comment:
I talked to Radim and he mentioned that he is not sure if this ok. I agree that this could problem in future if we eventually add more options to builder. @Ladicek wdyt?